### PR TITLE
*: fix setval in another session return num when it is already satisfied in meta

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?
fix setval in another session return num when it is already satisfied in meta
```
session A:
create sequence seq
select setval(seq, 100)

session B start when session A done.
select setval(seq, 50)   // here will return 50, should return NULL cause it is already satisfied in meta
```

### What is changed and how it works?

change the rebase interface to return rebase state, not just return nil or error.

How it Works:

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -281,11 +286,13 @@ func (alloc *allocator) rebase4Sequence(tableID, requiredBase int64) error {
 		if alloc.sequence.Increment > 0 {
 			if currentEnd >= requiredBase {
 				// Required base satisfied, we don't need to update KV.
+				alreadySatisfied = true
```
Comment (djshow832) [inline meta/autoid/autoid.go]: When the sequence almost runs out, can users rebase to a smaller value?

File: ddl/sequence_test.go [hunk 2]
```diff
@@ -660,6 +660,19 @@ func (s *testSequenceSuite) TestSequenceFunction(c *C) {
...
+	s.tk.MustExec("create sequence seq")
+	s.tk.MustQuery("select setval(seq, 100)").Check(testkit.Rows("100"))
+	se, err := session.CreateSession4Test(s.store)
+	c.Assert(err, IsNil)
+	tk1 := testkit.NewTestKit(c, s.store)
+	tk1.Se = se
+	tk1.MustExec("use test")
+	tk1.MustQuery("select setval(seq, 50)").Check(testkit.Rows("<nil>"))
+	tk1.MustQuery("select setval(seq, 100)").Check(testkit.Rows("<nil>"))
+	tk1.MustQuery("select setval(seq, 101)").Check(testkit.Rows("101"))
```
Comment (djshow832) [inline ddl/sequence_test.go]: Also check the real sequence value by selecting `nextval()`.
And check the situation when increment < 0.

File: meta/autoid/autoid.go (hunk 1 above)
Comment (AilinKid) [reply meta/autoid/autoid.go]: nop

File: ddl/sequence_test.go (hunk 2 above)
Comment (AilinKid) [reply ddl/sequence_test.go]: ok

Comment (AilinKid) [reply ddl/sequence_test.go]: Addressed @djshow832

File: meta/autoid/autoid.go [hunk 3]
```diff
@@ -299,9 +306,12 @@ func (alloc *allocator) rebase4Sequence(tableID, requiredBase int64) error {
...
 	metrics.AutoIDHistogram.WithLabelValues(metrics.TableAutoIDRebase, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
 	if err != nil {
-		return err
+		return 0, false, err
 	}
-	return nil
+	if alreadySatisfied {
+		return 0, true, nil
+	}
+	return requiredBase, false, nil
```
Comment (zimulala) [inline meta/autoid/autoid.go:314]:
```suggestion
	return requiredBase, false, err
```
Then we can remove line308-310

File: table/tables/tables.go [hunk 4]
```diff
@@ -1367,15 +1372,17 @@ func (t *TableCommon) SetSequenceVal(ctx interface{}, newVal int64, dbName, seqN
...
 	if err != nil {
 		return 0, false, err
 	}
-	// write sequence binlog to the pumpClient.
-	if ctx.(sessionctx.Context).GetSessionVars().BinlogClient != nil {
-		err = writeSequenceUpdateValueBinlog(ctx.(sessionctx.Context), dbName, seqName, seq.end)
-		if err != nil {
-			return 0, false, err
+	if !alreadySatisfied {
+		// write sequence binlog to the pumpClient.
```
Comment (zimulala) [inline table/tables/tables.go:1380]:
```suggestion
		// Write sequence binlog to the pumpClient.
```

**Pull request merged**